File: Counting-Sort/js/simple.js
const arr = []

function initData(count){
    for(let i = 0 ; i < count ; i++){
        arr.push(Math.floor(Math.random()*50))
    }
}

initData(10)

console.log(arr)


function sort(){
    let maxVal

    arr.forEach((v,i) => {
        if(i === 0){
            maxVal = v
        }
        if(v > maxVal){
            maxVal = v
        }
    })

    return countingSort(arr,maxVal)
}

function countingSort(arr,maxVal){
    const bucket = new Array(maxVal+1)
    let sortedIdx = 0
    
    for(let i = 0 ; i < arr.length ; i++){
        if(!bucket[arr[i]]){
            bucket[arr[i]] = 1
        }else{
            bucket[arr[i]]++
        }
    }

    for(let j = 0 ; j < bucket.length; j++){
        while(bucket[j] > 0){
            arr[sortedIdx] = j;
            sortedIdx++
            bucket[j]--
        }
    }

    return arr
}

console.log(sort())

File: Counting-Sort/python/simple.py
import random

arr = [random.randint(-20,50) for _ in range(6)]

print(arr)

def sort():
    max = -1;
    for i in range(len(arr)):
        if i == 0 :
            max = arr[i]
        
        if arr[i] > max :
            max = arr[i]

    res = countingSort(arr,max)
    print(res)

def countingSort(arr,max):

    bucket = [0]*(max+1)
    idx = 0
    
    for i in range(len(arr)):
        if not bucket[arr[i]]:
            bucket[arr[i]] = 1
        else:
            bucket[arr[i]] += 1

    for j in range(max+1):
        while bucket[j]:
            arr[idx] = j
            idx+=1
            bucket[j] -= 1

    return arr


sort()